### PR TITLE
feat: add geag client

### PR DESCRIPTION
**Pull request by long74100**
https://2u-internal.atlassian.net/browse/ENT-6090

**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: get_smarter_api_clients/geag.py [hunk 1]
```diff
@@ -2,9 +2,25 @@
 Client for GetSmarter API Gateway.
 """
 
+from get_smarter_api_clients.oauth import OAuthApiClient
```
Comment (adamstankiewicz) [inline get_smarter_api_clients/geag.py]: nit: should this be `getsmarter_api_clients.oauth` (and in the directory name)?

File: get_smarter_api_clients/geag.py [hunk 2]
```diff
@@ -2,9 +2,25 @@
...
 
-def stub():
+
+class GetSmarterEnterpriseApiClient(OAuthApiClient):
     """
-    Stub method for initial testing.
+    Client to interface with the GetSmarter Enterprise API Gateway(GEAG).
```
Comment (adamstankiewicz) [inline get_smarter_api_clients/geag.py]: nit: add space between `Gateway` and `(GEAG)`

File: get_smarter_api_clients/geag.py [hunk 3]
```diff
@@ -2,9 +2,25 @@
...
     """
-    print("stub")
+
+    def get_terms_and_conditions(self):
+        """
+        Fetch and return the terms and conditions from GEAG.
```
Comment (adamstankiewicz) [inline get_smarter_api_clients/geag.py]: nit: Do you think its worthwhile to expand on this docstring to explicitly include the 4 things it returns (i.e., it goes beyond just terms and conditions)?

* Student Terms and Conditions
* Website Terms of Use
* Privacy Policy
* Cookie Policy

File: tests/get_smarter_api_clients/test_oauth.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import ddt
+import pytz
+import responses
+
+from get_smarter_api_clients.oauth import OAuthApiClient
+
+
+class BaseOAuthApiClientTests(TestCase):
+    """
+    Base class with
```
Comment (christopappas) [inline tests/get_smarter_api_clients/test_oauth.py]: don't leave me hanging!

**Review by christopappas — APPROVED** [review]

File: tests/get_smarter_api_clients/test_oauth.py (hunk 4 above)
Comment (long74100) [reply tests/get_smarter_api_clients/test_oauth.py]: oh no

File: get_smarter_api_clients/geag.py (hunk 1 above)
Comment (long74100) [reply get_smarter_api_clients/geag.py]: yeah getsmarter is one word 😵‍💫

File: get_smarter_api_clients/geag.py (hunk 3 above)
Comment (long74100) [reply get_smarter_api_clients/geag.py]: renamed to `get_terms_and_policies`

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**